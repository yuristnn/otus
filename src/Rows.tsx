import React from 'react';
import Cell from './Cell';

type RowsProps = {
    rows: number;
    cells: number;
};

const Rows: React.FC<RowsProps> = (props) => {
    const { rows, cells } = props;

    const arrRows = Array.from(Array(rows).keys());
    const arrCells = Array.from(Array(cells).keys());
    return (
        <>
            {arrRows.map((row) => (
                <tr key={row}>
                    {arrCells.map((cell) => (
                        <Cell num={row * cells + cell + 1} key={cell} />
                    ))}
                </tr>
            ))}
        </>
    );
};

export default Rows;
