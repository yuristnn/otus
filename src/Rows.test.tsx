import 'jsdom-global/register';
import React from 'react';
import Rows from './Rows';
import { mount, shallow } from 'enzyme';

describe('Rows component', () => {
    it('Should render Rows component', () => {
        const component = shallow(<Rows rows={3} cells={3} />);
        expect(component).toMatchSnapshot();
    });

    it('Checking rows and cells', () => {
        const wrapper = mount(
            <table>
                <tbody>
                    <Rows rows={3} cells={3} />
                </tbody>
            </table>
        );
        expect(wrapper.find('tr')).toHaveLength(3);
        expect(wrapper.find('td')).toHaveLength(9);
    });
});
