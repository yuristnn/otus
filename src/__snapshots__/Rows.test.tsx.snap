// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rows component Should render Rows component 1`] = `ShallowWrapper {}`;
